packages/bp/e2e/admin/bots.test.ts

```diff
@@ -79,65 +76,67 @@ describe('Admin - Bot Management', () => {
   it('Export bot', async () => {
     await clickButtonForBot('#btn-export', tempBotId)
 
-    const response = await page.waitForResponse(`${bpConfig.host}/api/v2/admin/workspace/bots/${tempBotId}/export`)
-    expect(response.status()).toBe(200)
+    const response = await expectAdminApiCallSuccess(`workspace/bots/${tempBotId}/export`, 'GET')
 
     const responseSize = Number(response.headers()['content-length'])
     expect(responseSize).toBeGreaterThan(100)
   })
 
   it('Create revision', async () => {
-    await Promise.all([
-      expectAdminApiCallSuccess(`workspace/bots/${tempBotId}/revisions`, 'POST'),
-      clickButtonForBot('#btn-createRevision', tempBotId)
-    ])
-    await closeToaster()
+    await clickButtonForBot('#btn-createRevision', tempBotId)
+
+    await expectAdminApiCallSuccess(`workspace/bots/${tempBotId}/revisions`, 'POST')
   })
 
   it('Rollback revision', async () => {
+    // Super Hack to make sure the revision is 'ready'
+    await page.waitFor(500)
```

Comment:
For some obscure reason, I haven't found an alternative solution to this hackish 'solution'. It seems to make this test more stable that way (otherwise, it would sometimes fail locally for no explicit reason).

Reply:
This test doesn't wait for any of the elements to exists, so it might be possible that the buttons haven't been rendered yet, and the sleep is giving the page enough time to draw? other tests do a `waitForSelector`, that might also help here instead of sleeping but I'm not sure

Reply:
I've used this kind of hack in integration tests before and it works, until it doesn't. I don't have any solid lead on how to solve the problem because I haven't played with crawlers before, but maybe I can suggest this:

- As long as it doesn't break the function or component, and considering it's a valid option, could we not send a callback that's executed when the component is rendered? In this callback we could execute the test. It seems like another hacky solution but the added benefit would be that the test-execution-readiness won't ever be decided by an arbitrary `500` miliseconds, but rather by the callback being called, whenever it's ready. Not even sure it's viable.